lib/ActivityStreams.js

```diff
@@ -0,0 +1,144 @@
+/* globals NewTabURL, EventEmitter, XPCOMUtils */
+
+"use strict";
+
+const {Cu} = require("chrome");
+const {data} = require("sdk/self");
+const {PageMod} = require("sdk/page-mod");
+const {PlacesProvider} = require("lib/PlacesProvider");
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource:///modules/NewTabURL.jsm");
+
+XPCOMUtils.defineLazyGetter(this, "EventEmitter", function () {
+  const {EventEmitter} = Cu.import("resource://devtools/shared/event-emitter.js", {});
+  return EventEmitter;
+});
+
+let ActivityStreams = function (options) {
+  console.debug(`ActivityStreams.init`);
+
+  this.options = options ? options : {};
+  this.pageURL = this.options.pageURL ? options.pageURL : data.url("content/activity-streams.html");
+
+  this.workers = new Set();
+  EventEmitter.decorate(this);
+
+  this._setupPageMod();
+  this._setupListeners();
+  NewTabURL.override(this.pageURL);
+};
+
+ActivityStreams.prototype = {
+
+  pageMod: null,
+
+  /**
+   * Send a message to a worker
+   */
+  send(actionName, msg, worker) {
```

Comment:
.send() seems like indirection to me. That is, we are basically receiving 3 arguments and adding "addon-to-content". An alternative would be to just expose "addon-to-content" as a static and let the caller just call:

```JS
worker.port.emit(ActivityStreams.SEND, {type, data});
```